### ISSUE TITLE
false positive: variable with type None not valid as type

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#9563]
Is this a defect? Seems like a defect.
```py
from typing import TypedDict

none: None = None
a = TypedDict('a', {
    "foo": None,
    "bar": none,  # error: Variable "foo.none" is not valid as a type
})
```

**Comment by gvanrossum** [comment]
Wrong, ‘none’ is not a type, it’s a value of type NoneType.

**Comment by KotlinIsland** [comment]
But it can act as a type: `"foo": None`
